### PR TITLE
Fixes #16917: Permissions dirs recursive doesn't allow to set only owner, or group, or mode

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/16917

### CODE REVIEW

**Comment by ncharles** [comment]
Commit modified

File: tree/20_cfe_basics/ncf_lib.cf [hunk 1]
```diff
@@ -1459,3 +1459,20 @@ body acl add_posix_acl(aces)
...
       aces       => { @{aces} };
 }
+
+
+body perms ncf_mog_with_option(mode,user,group)
+# @brief Set the file's mode, owner and group
+# @param mode The new mode
+# @param user The username of the new owner
+# @param group The group name
+# Uses the classes permission_type_mode_set, permission_type_owner_set, permission_type_group_set to guard changing perms
```
Comment (Fdall) [inline tree/20_cfe_basics/ncf_lib.cf]: "Nice" cfengine trick

File: tests/acceptance/30_generic_methods/permissions_dirs_recursive.cf [hunk 2]
```diff
@@ -50,6 +79,7 @@ bundle agent test
 {
   methods:
     "ph1" usebundle => permissions_dirs_recursive("${init.directory}", "${init.mode}", "${init.owner}", "${init.group}");
+    "ph2" usebundle => permissions_dirs_recursive("${init.directory2}", "${init.mode2}", "", "${init.group2}");
```
Comment (Fdall) [inline tests/acceptance/30_generic_methods/permissions_dirs_recursive.cf]: Could you change the test to remove the mode and only change the owner or group?
Or add a new one.
The current one changes 2 things at the same time and make it unclear on its behaviour

Comment (ncharles) [reply tests/acceptance/30_generic_methods/permissions_dirs_recursive.cf]: ok

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
OK, squash merging this PR

**Comment by ncharles** [comment]
OK, squash merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**